### DISCUSSION TITLE
Adding logs to mocha awesome report and retrieving test context

### DISCUSSION THREAD

**Discussion by vinayak-kaladhar** [pactumjs/pactum#107] [Q&A] [answered]
**Question**: 
1. Am not able to retrieve the test context (for example: test title etc.)
tried adding addContext(this, '') as per mocha documentation, but did not work
2. How to pass such contexts from pactum to mocha-awesome resports?
3. How to add extra logs to our test case? other than using: settings.setLogLevel('ERROR')

```js
settings.setLogLevel('ERROR');
it('Verify creation of surgery facility without surgery related fields ', async () => {
    console.log("request" + request.createFacilityWithRequiredFields.rates.CNA)
    request.createFacilityWithRequiredFields.salesforceID = configData.salesforceid + utility.randomInteger().toString()
    await pactum.spec()
        .post(endpoints.baseUrl + endpoints.createHCF)
        .withBody(request.createSurgeyFacilityWithoutRequiredFields)
        .withHeaders('Accept', 'application/json')
        .withHeaders('Authorization', configData.HCFAuthorization)
        console.log("request" + request.createFacilityWithRequiredFields.rates.CNA)
});
```

**Comment by ASaiAnudeep** [comment]
> **1. Am not able to retrieve the test context (for example: test title etc.)**

Passing test context only works with normal functions and won't work with arrow functions.

```js
it('works ', async function () { });

it('does not works ', async () => { });
```

> **2. How to pass such contexts from pactum to mocha-awesome resports?**

Can you give us more details with examples. 

> **3. How to add extra logs to our test case? other than using: settings.setLogLevel('ERROR')?**

What logs do you want to have visibility over?

**Reply by vinayak-kaladhar** [reply]
Passing test context only works with normal functions and won't work with arrow functions.
1.even on using -  it('works ', async function () { });
this.test.title does not provide me with test details - its name and spec()
2. The mocha awesome report does not print the settings.setLogLevel('ERROR') or any other failure logs - it has only the entire function()
3. How to do custom logging , like the one similar to java - Log.info("info message displayed")

**Reply by ASaiAnudeep** [reply]
Added as [issue](https://github.com/pactumjs/pactum-examples/issues/4) to track adding an example with mocha awesome reporter.

**Comment by ASaiAnudeep** [comment, marked as answer]
@vinayak-kaladhar - added an example of integration of pactum with mochawesome reporter [here](https://github.com/pactumjs/pactum-examples/tree/main/reporter-mochawesome).

> This example works from [v3.1.5](https://github.com/pactumjs/pactum/releases/tag/v3.1.5)
